ngrok/src/session.rs

```diff
@@ -198,24 +197,18 @@ where
 /// Discards any errors during reconnect, allowing attempts to recur
 /// indefinitely.
 pub async fn default_connect(
-    addr: String,
+    host: String,
+    port: u16,
     tls_config: Arc<rustls::ClientConfig>,
     _: Option<AcceptError>,
 ) -> Result<Box<dyn IoStream>, ConnectError> {
-    let mut split = addr.split(':');
-    let host = split.next().unwrap();
-    let port = split
-        .next()
-        .map(str::parse::<u16>)
-        .transpose()?
-        .unwrap_or(443);
-    let conn = tokio::net::TcpStream::connect(&(host, port))
+    let conn = tokio::net::TcpStream::connect(&(host.as_str(), port))
         .await
         .map_err(ConnectError::Tcp)?
         .compat();
 
     let tls_conn = async_rustls::TlsConnector::from(tls_config)
-        .connect(rustls::ServerName::try_from(host)?, conn)
+        .connect(rustls::ServerName::try_from(host.as_str()).unwrap(), conn)
```

Comment:
Why the change to `unwrap` here?

Reply:
It's getting validated in the `server_addr` builder method. I really wanted to pass a pre-validated type, but for some reason none of the ones available allowed going from validated type back to a string, which is needed for the TCP connection :confused: 

I can add a comment to ward off future readers here

Reply:
Changed the unwrap to a clarifying `expect`